src/components/layouts/Default.tsx

```diff
@@ -58,11 +62,25 @@ interface Props extends CommonProps {
 export function Default(props: Props) {
   const [sidebarOpen, setSidebarOpen] = useState(false);
 
+  const whiteLabelLink = import.meta.env
+    .VITE_WHITELABEL_INVOICE_URL as unknown as string;
+
+  const account = useCurrentAccount();
+
+  const shouldShowUnlockButton =
+    (isHosted() &&
+      (freePlan() ||
+        (account?.plan && new Date(account?.plan_expires) < new Date()))) ||
+    (isSelfHosted() &&
+      ((account?.plan && new Date(account?.plan_expires) < new Date()) ||
+        !account?.plan));
+
```

Comment:
Is it possible to clean up this a bit? Perhaps extract it into a hook or something with simple ifs, it's really unreadable this way.

Reply:
The logic is separated into hooks.